hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -642,4 +647,9 @@ public SavepointStack savepointStack() {
     public void revertChildRecords() {
         recordListBuilder.revertChildrenOf(recordBuilder);
     }
+
+    public enum PrecedingTransactionCategory {
+        GENESIS,
```

Comment:
Since this is also used by `EndOfStakingPeriodUpdater`, and is used to decide whether we enforce the child records limit, maybe enum values like `LIMITED_CHILD_RECORDS` and `UNLIMITED_CHILD_RECORDS` would be more informative? 🤔

Reply:
Agreed. Changed those. Thanks @tinker-michaelj